routing/routing_tests/checkpoint_predictor_test.cpp

```diff
@@ -0,0 +1,145 @@
+#include "testing/testing.hpp"
+
+#include "routing/checkpoint_predictor.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+using namespace std;
+
+namespace
+{
+void TestAlmostEqual(double v1, double v2)
+{
+  double constexpr kEpsMeters = 1.0;
+  TEST(my::AlmostEqualAbs(v1, v2, kEpsMeters), (v1, v2));
+}
+}  // namespace
+
+namespace routing
+{
+class CheckpointPredictorTest
+{
+public:
+  CheckpointPredictorTest() : m_checkpointPredictor({0.0, 0.0} /* start */, {4.0, 0.0} /* finish */) {}
+
+  template <typename Fn>
+  size_t ForEachIntermediatePoint(vector<m2::PointD>::const_iterator beginIntermediatePoints,
+                                  vector<m2::PointD>::const_iterator endIntermediatePoints,
+                                  m2::PointD const & addingPoint, Fn && fn) const
+  {
+    return m_checkpointPredictor.ForEachIntermediatePoint(
+        beginIntermediatePoints, endIntermediatePoints, addingPoint, forward<Fn>(fn));
+  }
+
+  void FillCheckpoints(vector<m2::PointD>::const_iterator beginIntermediatePoints,
+                       vector<m2::PointD>::const_iterator endIntermediatePoints,
+                       vector<m2::PointD> & checkpoints) const
+  {
+    m_checkpointPredictor.FillCheckpoints(beginIntermediatePoints, endIntermediatePoints, checkpoints);
+  }
+
+  CheckpointPredictor m_checkpointPredictor;
```

Comment:
s/m_checkpointPredictor/m_predictor/

Please, keep your names as simple as it possible without lose of meaning.